tempodb/encoding/vparquet4/schema.go

```diff
@@ -547,6 +547,8 @@ func traceToParquetWithMapping(id common.ID, tr *tempopb.Trace, ot *Trace, dedic
 		}
 	}
 
+	// modify trace by rebatching spans and assigning nested set values
+	rebatchTrace(ot)
```

Comment:
should we also call this after combining here-ish?

https://github.com/grafana/tempo/blob/main/tempodb/encoding/vparquet4/combiner.go#L139

we seem to have three things we do every time we build a trace:
- rebatch
- nested set model
- service stats

should we create some holistic function like "finalizeTrace" that accomplishes this? that way if we add more of this kind of processing we will have an easy place to put it?

```
func finalizeTrace() {
  rebatch()
  assignNestedSetModel()
  ...
}
```